testing_app/README.md

```diff
@@ -0,0 +1,52 @@
+# testing_app
+
+A Sample app that shows different types of testing in Flutter.
+
+This particular sample uses the [Provider][] package but any other state management approach
+would do.
+
+[provider]: https://pub.dev/packages/provider
+
+## Goals for this sample
+
+Show how to perform:
+
+- Widget Testing,
+- Flutter Driver(Integration) Testing,
+- Performance Testing, and
+- State Management Testing using the [Provider][] package.
+
+## How to run tests
+- Navigate to the project's root folder using command line and follow the instructions below.
+
+### To run tests using only the Flutter SDK:
+The Flutter SDK can run some of the testing types on its own viz. unit testing and widget testing without the need of a physical device/emulator.
+- To run all the test files in the `test/` directory in one go, run `flutter test`.
+- To run a particular test file, run `flutter test test/<file_path>`
+
+### To run tests on a physical device/emulator:
```

Comment:
It might also make sense to add information about e2e test. I can make a PR to this after it lands, if that's easier. This is also relevant since in the next section it mentioned Firebase test lab, which is one thing e2e is trying to adapt to.

Reply:
It'd would be great if you can do that as I'm not that familiar with e2e tests as of now. I should probably start learning about it. =)